File: 2102-find-the-middle-index-in-array/2102-find-the-middle-index-in-array.java
class Solution {
    public int findMiddleIndex(int[] nums) {
        int n = nums.length, rightSum = 0, leftSum = 0;
        for (int num : nums) {
            rightSum += num;
        }
        for (int i = 0; i < n; i++) {
            if (rightSum - nums[i] == leftSum) {
                return i;
            }
            rightSum -= nums[i];
            leftSum += nums[i];
        }
        return -1;
    }
}

File: 2102-find-the-middle-index-in-array/2102-find-the-middle-index-in-array.py
class Solution:
    def findMiddleIndex(self, nums: List[int]) -> int:
        n = len(nums)
        rightSum = sum(nums)
        leftSum = 0
        for i in range(n):
            if rightSum - nums[i] == leftSum:
                return i
            rightSum -= nums[i]
            leftSum += nums[i]
        return -1